### PR TITLE
Implements verb name, category, and hidden & improves the interface

### PR DESCRIPTION
**Pull request by ike709**
Also makes ProcAttributes both work again and not be unnecessarily serialized.

I also improved the interface a bit and fixed the initial tab not being updated until after switching tabs,

Pics of the DMF improvements (most/all of those verbs don't actually have their name set, I promise it does work):
image
image

### CODE REVIEW

File: OpenDreamShared/Dream/Procs/ProcAttributes.cs [hunk 1]
```diff
@@ -6,7 +6,7 @@ namespace OpenDreamShared.Dream.Procs;
 [Flags]
 public enum ProcAttributes
 {
-    None = 0, // Internal
+    None = 1 << 0, // Internal
```
Comment (wixoaGit) [inline OpenDreamShared/Dream/Procs/ProcAttributes.cs:9]: Why was this changed to 1?

**Review by wixoaGit — CHANGES_REQUESTED** [review]

File: OpenDreamShared/Dream/Procs/ProcAttributes.cs (hunk 1 above)
Comment (ike709) [reply OpenDreamShared/Dream/Procs/ProcAttributes.cs:9]: Because it made the default value setting serialize it properly.

**Review by wixoaGit — APPROVED** [review]
